baremetal/metal3machine_manager.go

```diff
@@ -812,36 +817,46 @@ func (m *MachineManager) chooseHost(ctx context.Context) (*bmh.BareMetalHost, *p
 		if host.Spec.ConsumerRef != nil ||
 			(m.nodeReuseLabelExists(ctx, &host) &&
 				!m.nodeReuseLabelMatches(ctx, &host)) {
+			m.Log.Info(fmt.Sprintf("Continue was hit while checking ConsumerRef %v on host %v", host.Spec.ConsumerRef, host.Name))
 			continue
 		}
 		if host.GetDeletionTimestamp() != nil {
+			m.Log.Info("Continue was hit, host DeletionTimestamp is", "host", host.Name, "GetDeletionTimestamp", host.GetDeletionTimestamp)
 			continue
 		}
 		if host.Status.ErrorMessage != "" {
+			m.Log.Info("Continue was hit, host status ErrorMessage is not nil", "host", host.Name, "ErrorMessage", host.Status.ErrorMessage)
 			continue
 		}
 		switch host.Status.Provisioning.State {
 		case bmh.StateReady, bmh.StateAvailable:
```

Comment:
Can we add a log here to check the host state in case it is not ready or available

Reply:
Yeap, added.